doc/api/next_api_changes/deprecations.rst

```diff
@@ -135,3 +135,8 @@ access the transform classes from the :mod:`.scale` module.
 ``TexManager.cachedir``
 ~~~~~~~~~~~~~~~~~~~~~~~
 Use `matplotlib.get_cachedir()` instead.
+
+``OldScalarFormatter``, ``IndexFormatter`` and ``DateIndexFormatter``
```

Comment:
I don't see the deprecation for OldScalarFormatter.  While you are at it, maybe also deprecate OldAutoLocator.  I think the renaming of these things to "Old" was probably done with eventual deprecation and removal in mind.  The IndexFormatters also seem like good candidates for the trash bin.  I can imagine NullFormatter being useful in a user interface as one of a choice of Formatters, but it's true, it can be trivially recreated directly or via FuncFormatter.

Reply:
Oops, added it for OldScalarFormatter.  Also deprecated OldAutoLocator.
NullFormatter is trivial to recreate with FuncFormatter (or even simpler, `FormatStrFormatter("")` :-)), but I think it's widely used enough and its implementation so simple that we can leave it in, at least for now.